src/shared/logs-show.c

```diff
@@ -1595,21 +1611,36 @@ int add_matches_for_unit(sd_journal *j, const char *unit) {
         return r;
 }
 
-int add_matches_for_user_unit(sd_journal *j, const char *unit, uid_t uid) {
+int add_matches_for_user_unit_full(sd_journal *j, sd_bus *bus, const char *unit, uid_t uid, bool add_invocation) {
+        sd_id128_t id;
         int r;
 
         assert(j);
         assert(unit);
 
+        if (add_invocation) {
+                assert(bus);
+
+                r = unit_get_invocation_id(bus, unit, &id);
+                if (r < 0)
+                        return r;
+                if (r == 0) {
+                        log_debug("No invocation ID found for %s.", unit);
+                        add_invocation = false;
+                }
+        }
```

Comment:
so far this function was entirely "offline", which is a nice property. I think adding the invocation match here makes a ton of sense, but I'd prefer if this was done in a separate function, i.e. add_matches_for_unit_current_invocation() or so.